cmd/ocagent/config.go

```diff
@@ -102,7 +126,72 @@ func parseOCAgentConfig(yamlBlob []byte) (*config, error) {
 	return &cfg, nil
 }
 
-type exporterParser func(yamlConfig []byte) (te []exporter.TraceExporter, err error)
+// The goal of this function is to catch logical errors such as
+// if the Zipkin interceptor port conflicts with that of the exporter
+// lest we'll have a self DOS because spans will be exported "out" from
+// the exporter, yet be received from the interceptor, then sent back out
+// and back in a never ending loop.
+func (c *config) checkLogicalConflicts(blob []byte) error {
```

Comment:
Possibly a good candidate for backlog: it seems that the intent is for this to be a general check of consistency on the config but right now it just addresses the zipkin interceptor (unique case so far I guess). We are also doing the unmarshal in various places (while trivial it is nice to avoid it if we can), and as it grows it perhaps deserves its own package so it can be easily shared.

If others agree we can add this to the backlog. /cc @flands

Reply:
Make sense to me, please file an issue to keep track of this.

Reply:
Done, here it is issue https://github.com/census-instrumentation/opencensus-service/issues/135

Reply:
Thanks for filing it @pjanotti I was thinking the same thing a couple of days back but was stuck at a conference.